### PR TITLE
Makes chem dispenser recharging far less glacial

### LINKED ISSUE #36687: Chemical Dispenser power regenerates GLACIALLY slowly
Issue reported from  Round ID: 85472 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])

Reporting client version: 512

I have no idea how it actually regenerates, but it's too slow even before considering that t4 cell gives it 8000 points.

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Fixes #36687

So yeah turns out I chopped off half the recharging bonuses while also reducing the effective recharge amount into 1/10. This should be much better

:cl: Naksu
fix: Chem dispensers no longer take hours to recharge
tweak: chem dispensers also now use energy from the grid in proportion with the amount charged
/:cl:

### CODE REVIEW

**Comment by newfren** [comment]
how does this compare to how it was before the parts change, both at roundstart and at tier 4 parts?

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 1]
```diff
@@ -72,11 +71,15 @@
 	return ..()
 
 /obj/machinery/chem_dispenser/process()
-	if(recharged < 0)
```
Comment (Jalleo) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: Shouldn't this be a timer instead?

**Comment by improvedname** [comment]
Not forgetting you can deconstruct and reconstruct for instant recharge lol

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: we already have process(), we don't need a timer

File: code/modules/reagents/chemistry/machinery/chem_dispenser.dm [hunk 2]
```diff
@@ -72,11 +71,15 @@
...
-		recharge()
-		recharged = recharge_delay
-	else
-		recharged -= 1
+	if (recharge_counter >= 4)
+		if(stat & (BROKEN|NOPOWER))
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/machinery/chem_dispenser.dm]: !is_operational()

**Review by Cyberboss — DISMISSED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]
I'm hia f

**Pull request merged**